sfge/src/main/java/com/salesforce/rules/UnnecessarilyExtensibleClassRule.java

```diff
@@ -0,0 +1,98 @@
+package com.salesforce.rules;
+
+import com.salesforce.apex.jorje.ASTConstants;
+import com.salesforce.graph.Schema;
+import com.salesforce.graph.vertex.SFVertexFactory;
+import com.salesforce.graph.vertex.UserClassVertex;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.tinkerpop.gremlin.process.traversal.P;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.__;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+
+public class UnnecessarilyExtensibleClassRule extends AbstractStaticRule {
+    private static final String URL =
+            "https://forcedotcom.github.io./sfdx-scanner/en/v3.x/salesforce-graph-engine/rules/#UnnecessarilyExtensibleClassRule";
+
+    private static final String DESCRIPTION =
```

Comment:
@teresa-allen-sfdc , I'll need your feedback on this description, which will be used for `scanner:rule:describe` and in the public facing docs.

Reply:
`UserFacingMessages` is meant to be a placeholder for a config-based messages system. Can we keep the violation messages and description messages there and invoke the constants here?